### PR TITLE
add erc20 transfer proposal

### PR DESCRIPTION
**Pull request by ngmachado**
This PR closes issue #514

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,6 +183,23 @@ This script extracts the proposal id and queues the proposal via the OD governan
 
 The script extracts the necessary execution params from the JSON-the same params used during the proposal and executes the proposal. This script can be used arbitrarily for any proposal.
 
+##### ERC20 Transfer
```
Comment (pi0neerpat) [inline README.md:186]: Excellent thank you!

File: script/testScripts/gov/ERC20TransferAction/ProposeERC20Transfer.s.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    bytes[] memory calldatas = new bytes[](1);
+    calldatas[0] = abi.encodeWithSelector(IERC20.transferFrom.selector, fromAddress, receiverAddress, amountToTransfer);
+
+    // Get the description and descriptionHash
+    string memory description = 'Transfer ERC20 tokens to receiver';
+    bytes32 descriptionHash = keccak256(bytes(description));
+
+    vm.startBroadcast(vm.envUint('GOV_EXECUTOR_PK'));
+
+    // Propose the action to add the collateral type
```
Comment (pi0neerpat) [inline script/testScripts/gov/ERC20TransferAction/ProposeERC20Transfer.s.sol:46]: I think this comment is old

File: script/testScripts/gov/ERC20TransferAction/ProposeERC20Transfer.s.sol [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    //   "proposalId": uint256,
+    //   "tokenAddress": string,
+    //   "fromAddress": string,
+    //   "toAddress": string,
+    //   "amountToTransferInWei": string,
+    //   "targets": address[],
+    //   "values": uint256[]
+    //   "calldatas": bytes[],
+    //   "description": string,
+    //   "descriptionHash": bytes32,
```
Comment (pi0neerpat) [inline script/testScripts/gov/ERC20TransferAction/ProposeERC20Transfer.s.sol:80]: Great work. Do you have any suggestions for how delegates and community members can verify the contract call data in proposals? This is especially helpful for non solidity developers. We want everyone to be able to make informed decisions. For example, here is a recent HAI proposal on Tally, which I can't expect anyone other than seasoned web3 devs to interpret 

image


Here's the recent discussion in HAI's Discord on the subject:  https://discord.com/channels/1110925389464883201/1206632791513759825/1217556013151162410

**Review by pi0neerpat — APPROVED** [review]

File: script/testScripts/gov/ERC20TransferAction/ProposeERC20Transfer.s.sol (hunk 3 above)
Comment (ngmachado) [reply script/testScripts/gov/ERC20TransferAction/ProposeERC20Transfer.s.sol:80]: In web3 governance, we're tackling two big issues:

**Checking if the Code is Correct:** (_Valid Code_) We've got tools like Tally and Tenderly for this. Tally runs the code to see if it does what it's supposed to, which helps us trust the code more.
**Making Sure the Code Matches Its Goal:** (_Fit for propose_) This part's trickier. Sometimes, a proposal says it will do one thing, but the actual code does something different. It's vital that not just coders, but everyone, can understand what a proposal's code is really about.

To bridge the gap for non-developers, we can create a companion webpage for governance app. We have a predefine set of governance actions that are valid and fit for propose.

This page would help explain the actions proposed in a way that's easy to understand. If a proposal can't be explained by this page (are not from our action library), that's a signal for us to take a closer look before moving forward.

Process workflow could be something like: Automatically connect Tenderly's code-checking tool to our explanatory companion page. Whenever a proposal is made, Tenderly would check the code, and our page would translate what each part means in plain English. Proposals that can't be decoded this way would be flagged as "needs review."

This should be a broader discussion, I will copy this text and added to discord.
